pkgs/by-name/li/libfprint-2-tod1-broadcom/package.nix

```diff
@@ -0,0 +1,89 @@
+{
+  autoPatchelfHook,
+  fetchzip,
+  lib,
+  libfprint-tod,
+  openssl,
+  patchelfUnstable,
+  stdenv,
+}:
+
+# Based on ideas from (using a wrapper library to redirect fopen() calls to firmware files):
+#   * https://tapesoftware.net/replace-symbol/
+#   * https://github.com/NixOS/nixpkgs/pull/260715
+let
+  version = "5.12.018";
+
+  unpacked = stdenv.mkDerivation {
+    pname = "libfprint-2-tod1-broadcom-unpacked";
+    inherit version;
+
+    src = fetchzip {
+      url    = "http://dell.archive.canonical.com/updates/pool/public/libf/libfprint-2-tod1-broadcom/libfprint-2-tod1-broadcom_5.12.018.orig.tar.gz";
+      sha256 = "sha256-0C2PpYpEJNrU+8NT95w4QV0J5nHQisMY94Czw3jQOzw=";
+    };
+
+    dontBuild = true;
+
+    installPhase = ''
+      mkdir $out
+      cp -r . $out
+    '';
+  };
+
+  # wraps `fopen()` for finding firmware files
+  wrapperLibName = "wrapper-lib.so";
+  wrapperLib = stdenv.mkDerivation {
+    pname = "libfprint-2-tod1-broadcom-wrapper-lib";
+    inherit version;
+
+    src = ./.;
+
+    postPatch = ''
+      substitute wrapper-lib.c lib.c \
+        --subst-var-by to "${unpacked}/var/lib/fprint/fw"
+      cc -fPIC -shared lib.c -o ${wrapperLibName}
+    '';
+
+    installPhase = ''
+      install -D -t $out/lib ${wrapperLibName}
+    '';
+  };
+in stdenv.mkDerivation {
+  pname = "libfprint-2-tod1-broadcom";
+  inherit version;
+
+  src = fetchzip {
+    url    = "http://dell.archive.canonical.com/updates/pool/public/libf/libfprint-2-tod1-broadcom/libfprint-2-tod1-broadcom_5.12.018.orig.tar.gz";
+    sha256 = "sha256-0C2PpYpEJNrU+8NT95w4QV0J5nHQisMY94Czw3jQOzw=";
+  };
+
+  buildInputs       = [ libfprint-tod openssl wrapperLib ];
+  nativeBuildInputs = [ autoPatchelfHook ];
+  dontConfigure     = true;
+
+  installPhase = ''
+    install -D -m 644 -v LICENCE.broadcom "$out/share/licenses/$pname/LICENSE"
```

Comment:
do we really need the LICENSE file on the installed derivation?

Reply:
I guess you're right, I just included it since it was also installed in the package file from some other Linux distribution (don't remember which one). Anyway, I removed it.